src/core/agent/services/signifyNotificationService.ts

```diff
@@ -166,7 +166,44 @@ class SignifyNotificationService extends AgentService {
     if (notif.r) {
       return;
     }
-
+    if (notif.a.r === NotificationRoute.ExnIpexGrant) {
+      const exchange = await this.props.signifyClient
+        .exchanges()
+        .get(notif.a.d);
+      const existingCredential = await this.props.signifyClient
+        .credentials()
+        .get(exchange.exn.e.acdc.d)
+        .catch(() => undefined);
```

Comment:
This catch should be specific for the 404 - otherwise something else could've happened

Reply:
The keria is throwing 500 in this case at the moment, but I think that status is not quite right. What if they change it later to 404?

Reply:
What's printed in the KERIA logs?

Reply:
just "500 internal server error"

Reply:
ok its ok for now I've made a note on backlog